chemprop/v2/cli/predict.py

```diff
@@ -199,81 +197,61 @@ def main(args):
-            target_columns=args.target_columns,
-            p_features=args.cal_features_path,
-            p_atom_feats=args.cal_atom_features_path,
-            p_bond_feats=args.cal_bond_features_path,
-            p_atom_descs=args.cal_atom_descriptors_path,
-            **featurization_kwargs,
-        )
-        logger.info(f"calibration size: {len(cal_data)}")
-    else:
-        cal_data = None
+    # TODO: add uncertainty and calibration
+    # if args.cal_path is not None:
+    #     cal_data = build_data_from_files(
+    #         args.cal_path,
+    #         **format_kwargs,
+    #         target_columns=args.target_columns,
+    #         p_features=args.cal_features_path,
+    #         p_atom_feats=args.cal_atom_features_path,
+    #         p_bond_feats=args.cal_bond_features_path,
+    #         p_atom_descs=args.cal_atom_descriptors_path,
+    #         **featurization_kwargs,
+    #     )
+    #     logger.info(f"calibration size: {len(cal_data)}")
+    # else:
+    #     cal_data = None
 
     test_dset = make_dataset(test_data, bond_messages, args.rxn_mode)
 
-    test_loader = data.MolGraphDataLoader(test_dset, args.batch_size, args.n_cpu)
-    if cal_data is not None:
-        cal_dset = make_dataset(cal_data, bond_messages, args.rxn_mode)
-        cal_loader = data.MolGraphDataLoader(cal_dset, args.batch_size, args.n_cpu, shuffle=False)
-    else:
-        cal_loader = None
+    test_loader = data.MolGraphDataLoader(test_dset, args.batch_size, args.n_cpu, shuffle=False)
+    # TODO: add uncertainty and calibration
+    # if cal_data is not None:
+    #     cal_dset = make_dataset(cal_data, bond_messages, args.rxn_mode)
+    #     cal_loader = data.MolGraphDataLoader(cal_dset, args.batch_size, args.n_cpu, shuffle=False)
+    # else:
+    #     cal_loader = None
 
-    mp_kwargs = dict(
-        d_h=args.message_hidden_dim,
-        bias=args.message_bias,
-        depth=args.depth,
-        undirected=args.undirected,
-        dropout=args.dropout,
-        activation=args.activation,
-        aggregation=args.aggregation,
-        norm=args.norm,
-    )
-    mp_block = modules.molecule_block(*test_dset.featurizer.shape, bond_messages, **mp_kwargs)
-
-    extra_mpnn_kwargs = dict()
-    if args.dataset_type == "multiclass":
-        extra_mpnn_kwargs["n_classes"] = args.multiclass_num_classes
-    elif args.dataset_type == "spectral":
-        extra_mpnn_kwargs["spectral_activation"] = args.spectral_activation
-
-    mpnn_cls = get_mpnn_cls(args.dataset_type, args.loss_function)
-    model = mpnn_cls(
-        mp_block,
-        n_tasks,
-        args.ffn_hidden_dim,
-        args.ffn_num_layers,
-        args.dropout,
-        args.activation,
-        **extra_mpnn_kwargs,
-    )
     logger.info(model)
 
     trainer = pl.Trainer(
         logger=None,
         enable_progress_bar=True,
         accelerator="auto",
         devices=args.n_gpu if torch.cuda.is_available() else 1,
-        max_epochs=args.epochs,
     )
```

Comment:
wrap this whole block using [`torch.inference_mode()`](https://pytorch.org/docs/stable/generated/torch.inference_mode.html)

Reply:
@davidegraff is the right way to do this by putting the call to `trainer.predict` in a `with torch.inference_mode():` block (as [here](https://github.com/Lightning-AI/lightning/issues/11530#issuecomment-1067566245)), or to add a `inference_mode=False` argument to `pl.Trainer` (from the [documentation](https://lightning.ai/docs/pytorch/latest/common/trainer.html))?

I'm admittedly a bit confused by the documentation, since it says `Trainer(inference_mode=True)` is the default and `Trainer(inference_mode=False)` is equivalent to `torch.no_grad`, while I would expect the opposite to be true, unless I'm misunderstanding something...

Reply:
yeah that's a good question. Looking at the source, it doesn't seem like the argument actually does _anything_. Even if it does do something in the future, it won't hurt us if our code is already inside a `with`-block (i.e., the PyTorch docs)

Reply:
@kevingreenman By reading the documentation, I think `Trainer(inference_mode=True)` means that `torch.inference_mode():` is used by Trainer internally during model evaluation (i.e., when `Trainer.validate`, `Trainer.test`, and `Trainer.predict` are called), while `Trainer(inference_mode=False)` uses `torch.no_grad():` during model evaluation. But I would like to know @davidegraff's thoughts as well.